### PR TITLE
mpir: modernize

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **mpir/***

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by madebr — APPROVED** [review]

File: recipes/mpir/all/conanfile.py [hunk 1]
```diff
@@ -44,22 +44,25 @@ def configure(self):
             del self.settings.compiler.libcxx
             del self.settings.compiler.cppstd
 
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
     def build_requirements(self):
         if self.settings.compiler != "Visual Studio":
```
Comment (SpaceIm) [inline recipes/mpir/all/conanfile.py:52]:
```suggestion
        self.build_requires("yasm/1.3.0")
        if self.settings.compiler != "Visual Studio":
            self.build_requires("m4/1.4.19")
            if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
                self.build_requires("msys2/cci.latest")
```

File: recipes/mpir/all/conanfile.py [hunk 2]
```diff
@@ -44,22 +44,25 @@ def configure(self):
...
-            self.build_requires("m4/1.4.18")
+            self.build_requires("m4/1.4.19")
```
Comment (SpaceIm) [inline recipes/mpir/all/conanfile.py:53]:
```suggestion
```

File: recipes/mpir/all/conanfile.py [hunk 3]
```diff
@@ -44,22 +44,25 @@ def configure(self):
...
         self.build_requires("yasm/1.3.0")
```
Comment (SpaceIm) [inline recipes/mpir/all/conanfile.py:54]:
```suggestion
```

File: recipes/mpir/all/conanfile.py [hunk 4]
```diff
@@ -44,22 +44,25 @@ def configure(self):
...
-        if tools.os_info.is_windows and self.settings.compiler != "Visual Studio" and \
-           "CONAN_BASH_PATH" not in os.environ and tools.os_info.detect_windows_subsystem() != "msys2":
-            self.build_requires("msys2/20200517")
+        if self._settings_build.os == "Windows" and self.settings.compiler != "Visual Studio" and \
+           "CONAN_BASH_PATH" not in os.environ:
+            self.build_requires("msys2/cci.latest")
```
Comment (SpaceIm) [inline recipes/mpir/all/conanfile.py:57]:
```suggestion
```

**Review by SpaceIm — COMMENTED** [review]
factorize the logic in build_requirements

**Comment by SpaceIm** [comment]
Could you add `required_conan_version` please?

**Review by SpaceIm — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**